### DISCUSSION TITLE
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD with errorcode 134.

### DISCUSSION THREAD

**Discussion by smallriver666** [SeisSol/SeisSol#558] [Q&A] [answered]
I ran into this problem while running the example, and I googled to see that someone had this problem before and didn't know how to solve it in the end.
I installed it according to the following manual. 
https://seissol.readthedocs.io/en/latest/compilation.html
I don't know if it is because I changed the CPU architecture of the last -`DHOST_ARCH=skx` to `hsw`, because my computer is of hsw architecture.
![image](https://user-images.githubusercontent.com/104721358/170924690-688c02e0-4ba2-49b0-85dc-b1394ebe0b9f.png)

**Comment by Thomas-Ulrich** [comment]
try disabling the communication thread when compiling

**Reply by smallriver666** [reply]
> try disabling the communication thread when compiling

What can I do to disable the communication thread when compiling, when compiling the seissol software? What should I add. Because of the way of adding environment variables, I solved this problem under the centos7 system, but it was not solved in Ubuntu 18. The version in my ubuntu 18 is in the link below
https://github.com/SeisSol/Training/blob/main/Dockerfile
 The version in my centos7 is in the link below
https://seissol.readthedocs.io/en/latest/compilation.html
![image](https://user-images.githubusercontent.com/104721358/170999605-1a8715cc-2f13-4767-a5d4-6a50b125b644.png)
In the link https://seissol.readthedocs.io/en/latest/compilation.html, it didn't let me install `pumgen` and `gmsh`, so I tried to follow the link https://github.com/SeisSol/SeisSol/blob/master/.ci/docker/Dockerfile.base，In ubuntu, I did not encounter this problem, according to Dockerfile.base can be installed successfully
and tried to install pumgen and gmsh in `centos7`, but I encountered this problem again, I don't know what library I'm missing that can't find `APF`
I executed the following four executable programs, and solved three of them according to the error message. The last one generating_the_mesh.sh encountered an error. Among them, I have successfully installed gmsh, but pumgen encountered the following error.Is there some library I'm missing?

![image](https://user-images.githubusercontent.com/104721358/171001388-c2e1658c-1737-4778-824f-94dc72ea2acf.png)

**Reply by sebwolf-de** [reply]
Follow our documentation on how to build pumgen and its dependencies https://github.com/SeisSol/PUMGen/wiki/How-to-compile-PUMGen

**Comment by Thomas-Ulrich** [comment, marked as answer]
or use the following environment variables (with the executable compiled with comm thread on):
```
export MP_SINGLE_THREAD=no
unset KMP_AFFINITY
export OMP_NUM_THREADS=7
export OMP_PLACES="cores(7)"
```

**Reply by smallriver666** [reply]
> or use the following environment variables (with the executable compiled with comm thread on):
> 
> ```
> export MP_SINGLE_THREAD=no
> unset KMP_AFFINITY
> export OMP_NUM_THREADS=7
> export OMP_PLACES="cores(7)"
> ```

And I noticed that the environment variables required for different examples seem to be different

**Reply by sebwolf-de** [reply]
Yes, they depend on your computer. You have to set OMP_NUM_THREADS to the number of cores on your computer - 1

**Reply by smallriver666** [reply]
> Yes, they depend on your computer. You have to set OMP_NUM_THREADS to the number of cores on your computer - 1

Does this mean that all my examples do it this way, i.e. if I assign six cores to the virtual machine, I should do it.
```
export MP_SINGLE_THREAD=no
unset KMP_AFFINITY
export OMP_NUM_THREADS=5
export OMP_PLACES="cores(5)"
```

I mean `run all the examples` in the future like this, because I follow your prompt, it can also run in ubuntu.

**Reply by sebwolf-de** [reply]
Yes, for a six core machine this sound reasonable. If you have hyperthreading, you can also tweak the `cores(X)` statement, but in a virtual machine, it shouldn't make much of a difference.

**Comment by smallriver666** [comment]
thank you very much, I can solve this problem by using your method.
But I encountered another problem, that is, there is no valid parameter file. 
![image](https://user-images.githubusercontent.com/104721358/170951595-f8e833e6-95c9-40a9-b7b3-0958d49cc609.png)

My parameter file is copied from Examples in github, I don't know where the problem is, and I don't know if you know where the problem is. I really appreciate if you can help me.

**Reply by Thomas-Ulrich** [reply]
change PARAMETER with parameters.par

**Comment by smallriver666** [comment]
I don't know where to find the input `meshfile loh.puml.nrf`, I'm sorry to waste your time, my English is not good, I have a lot of difficulties in reading your instruction manual
![image](https://user-images.githubusercontent.com/104721358/170955265-c39f02a3-7205-45f0-bfce-8dbb4b9de06f.png)

**Reply by sebwolf-de** [reply]
`./generating_the_mesh.sh` creates the mesh file

**Reply by smallriver666** [reply]
> `./generating_the_mesh.sh` creates the mesh file

But the manual I installed did not mention `pumgen`. I encountered a problem installing `pumgen` in centos7, which is the question I asked above， can not find `APF`

**Reply by sebwolf-de** [reply]
https://github.com/SeisSol/PUMGen/wiki/How-to-compile-PUMGen

**Reply by smallriver666** [reply]
> https://github.com/SeisSol/PUMGen/wiki/How-to-compile-PUMGen
After I successfully installed pumgen, I encountered a problem again after running it, and I don't know why it happens.
![image](https://user-images.githubusercontent.com/104721358/171104954-15e4cdcb-0e43-42f1-92f6-e3d6849ef0f6.png)
![image](https://user-images.githubusercontent.com/104721358/171105035-70992ba9-3ba9-4c4d-b53d-c5d607160eda.png)

**Comment by sebwolf-de** [comment]
Probably, your mesh is too large, i.e. you try to allocate too much memory. Either you have to shrink your mesh (adjust element lenght in `.geo` scripts) or you have to use a bigger computer.

**Reply by smallriver666** [reply]
> Probably, your mesh is too large, i.e. you try to allocate too much memory. Either you have to shrink your mesh (adjust element lenght in `.geo` scripts) or you have to use a bigger computer.

I just switched to a bigger computer and maybe I ran into the same problem, malloc failed.
![image](https://user-images.githubusercontent.com/104721358/171116626-a29b9daf-4b80-4784-8091-87d27d7c02cd.png)


Maybe they are the same problem, so I want to know exactly how to change this file `.geo` scripts to modify the mesh.
is this file below? I'm sorry, I'm just starting to learn and I can't understand many parameters. what should I do to change this file.
![image](https://user-images.githubusercontent.com/104721358/171117858-dc4ab7f6-0de6-4f29-8ac6-5131bffddbff.png)
![image](https://user-images.githubusercontent.com/104721358/171117965-5028a69b-1917-44fd-8ec7-9ff2dfeb0f1b.png)

**Reply by smallriver666** [reply]
> Probably, your mesh is too large, i.e. you try to allocate too much memory. Either you have to shrink your mesh (adjust element lenght in `.geo` scripts) or you have to use a bigger computer.

Maybe what is running on the server can solve this problem?
